app/upgrades/v1rc7/testnet/upgrade.go

```diff
@@ -0,0 +1,101 @@
+package testnet
+
+import (
+	"context"
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/babylonlabs-io/babylon/app/keepers"
+	"github.com/babylonlabs-io/babylon/app/upgrades"
+	btcstakingkeeper "github.com/babylonlabs-io/babylon/x/btcstaking/keeper"
+	btcstakingtypes "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+)
+
+const (
+	UpgradeName = "v1rc7"
+)
+
+func CreateUpgrade() upgrades.Upgrade {
+	return upgrades.Upgrade{
+		UpgradeName:          UpgradeName,
+		CreateUpgradeHandler: CreateUpgradeHandler(),
+	}
+}
+
+// CreateUpgradeHandler upgrade handler for launch.
+func CreateUpgradeHandler() upgrades.UpgradeHandlerCreator {
+	return func(mm *module.Manager, cfg module.Configurator, keepers *keepers.AppKeepers) upgradetypes.UpgradeHandler {
+		return func(context context.Context, _plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			ctx := sdk.UnwrapSDKContext(context)
+
+			logger := ctx.Logger().With("upgrade", UpgradeName)
+			logger.Info("migrating finality providers...")
+			if err := MigrateFinalityProviders(ctx, keepers.BTCStakingKeeper); err != nil {
+				return nil, fmt.Errorf("failed migrate finality providers: %w", err)
+			}
+			logger.Info("finality providers migration done!")
+
+			migrations, err := mm.RunMigrations(ctx, cfg, fromVM)
```

Comment:
I think it is customary to add ` mm.RunMigrations(ctx, cfg, fromVM)` as first thing in the handler. It does not make a difference it this particular case (we knot there is not migartion here), but eitherway lets add it as first thing.